Doc/library/sqlite3.rst

```diff
@@ -1872,7 +1872,10 @@ are vulnerable to `SQL injection attacks`_ (see the `xkcd webcomic
 <https://xkcd.com/327/>`_ for a humorous example of what can go wrong)::
 
    # Never do this -- insecure!
-   symbol = 'RHAT'
+   # An attacker can select all stocks by entering "' OR 1 = 1; --",
+   # or load any extension by entering "' OR load_extension('./any.so'); --"
+   # if extension loading is enabled.
```

Comment:
@CAM-Gerlach, do you think it is needed to explain why these input strings are dangerous, or should we leave that as an exercise for the reader?

Reply:
That would seem prudent to me, to be explicit about the specific security problem to ensure users actually understand it rather than cargo-culting, which can lead to mistakes down the line. It seems particularly advisable in this case, since the comments are plain text without proper formatting and `"` is used to delineate literals here, beginners might miss the key element that makes the attack happen—the `'` immediately following it—and need some effort to figure it out.

Reply:
Good points, CAM. I agree, we should put more effort into the explanation. Either show more explicitly what actually happens after the substitution has been done[^1], or provide more background in the text above the code block. We might also consider removing the XKCD reference.

Another thing I've been thinking about in this particular how-to, is that I believe we should start with how to do it right, and then either warn about injection atttacks towards the end of the how-to. Either that, or link to a (new) _explanation_ that talk about injection attacks. I need to sleep on what's best here, but this issue should not hold this PR; we can do this in follow-up PRs.

[^1]: using simulated REPL doctest code block

Reply:
> show more explicitly what actually happens after the substitution has been done

Good idea! Using simulated REPL makes this example explicit and easy to understand. Also, I don't think it is necessary to specifically mention `load_extension` in this example.

```
>>> # Never do this -- insecure!
>>> # An attacker can select all stocks with a crafted input.
>>> symbol = input()
' OR 1 = 1; --
>>> sql = "SELECT * FROM stocks WHERE symbol = '%s'" % symbol
>>> sql
"SELECT * FROM stocks WHERE symbol = '' OR 1 = 1; --'"
>>> cur.execute(sql)
```

Reply:
I'm not sure if you mean in this section, or that one—if the latter, perhaps, if for some reason injection in manual calls to the `load_extension` SQL method specifically is a particularly common attack (as opposed to any other), but otherwise I'm not sure it is appropriate in a reference there, which should be astute and with examples directly and tightly focused on illustrating the specific behavior described. Security is certainly a priority to warn about, but I'm a little confused why that method specifically would be called out...perhaps I'm misunderstanding what you mean here?

Reply:
For this section, I think the "select all stocks" example is fine. It is concise and easy to understand.

The `load_extension` injection attack is a nice example of how SQL functions can be used maliciously, so we can consider adding an example of that to the `Connection.enable_load_extension` reference; but that is not for this issue/PR.